tasks/lint_staged_files.js

```diff
@@ -14,20 +21,21 @@ export default function (grunt) {
     const eslintSourcePaths = grunt.config.get('eslint.options.paths');
     if (!eslintSourcePaths) grunt.fail.warn('eslint.options.paths is not defined');
 
-    const sourcePathRegexps = cli.resolveFileGlobPatterns(eslintSourcePaths)
-      .map(glob => minimatch.makeRe(glob));
+    const sourcePathGlobs = cli.resolveFileGlobPatterns(eslintSourcePaths);
 
     const files = grunt.config
     .get('filesToCommit')
     .filter(isStaged)
     .map(getFilename)
-    .map(file => resolve(root, file))
+    .map(file => relative(root, resolve(file))) // resolve to pwd, then get relative from the root
     .filter(file => {
-      if (!sourcePathRegexps.some(re => file.match(re))) {
+      if (!sourcePathGlobs.some(glob => minimatch(file, glob))) {
+        skippingFile(file, 'not selected by grunt eslint config');
```

Comment:
I'm on the fence about whether this should be a warning or not, but I'm cool with giving it a go to see it in action for awhile.

This does bring up a good question for me: why not eslint everything and then rely on the blacklisting in eslint for exceptions rather than the combined whitelist/blacklist approach we're using now?